src/main/java/com/iota/iri/network/TransactionRequester.java

```diff
@@ -20,6 +18,7 @@
 
     private static final Logger log = LoggerFactory.getLogger(TransactionRequester.class);
     private final Set<Hash> transactionsToRequest = new LinkedHashSet<>();
+    private final Set<Hash> recentlyRequestedTransactions = Collections.synchronizedSet(new HashSet<>());
```

Comment:
The other option would be
`ConcurrentHashMap.newKeySet()` which is more performant under heavy contention.

However, I don't think we have too much contention here, so you probably made the more performant choice :+1: